pyproject.toml

```
@@ -0,0 +1,94 @@
+[project]
+name = "mache"
+dynamic = ["version"]
+authors = [
+    { name="Xylar Asay-Davis", email="xylar@lanl.gov" },
+    { name="Althea Denlinger", email="althea@lanl.gov" }
+]
+description = "A package for providing configuration data related to E3SM supported machines."
+license = {file = "LICENSE"}
+readme = "README.md"
+requires-python = ">=3.8"
+classifiers = [
+    # these are only for searching/browsing projects on PyPI
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+
+    "License :: OSI Approved :: BSD License",
+    "Operating System :: OS Independent",
+]
+
+dependencies = [
+    "importlib_resources",
+    "jinja2",
+    "lxml",
+    "pyyaml",
+    "progressbar2",
+]
+
+[project.optional-dependencies]
+docs = [
+    # building documentation
+    "sphinx",
+    "sphinx_rtd_theme",
+    "myst-parser",
+    "sphinx-multiversion",
+    "rst-to-myst",
+]
+
+dev = [
+    # linting and testing
+    "pip",
+    "pytest",
+    "isort",
+    "flake8",
+    "mypy",
+    "pre-commit",
+]
+
+[tool.isort]
+multi_line_output = "3"
+include_trailing_comma = true
+force_grid_wrap = "0"
+use_parentheses = true
+line_length = "79"
+skip = "e3sm_diags/e3sm_diags_driver.py"
+
+[tool.mypy]
+python_version = "3.10"
+check_untyped_defs = true
+ignore_missing_imports = true
+warn_unused_ignores = true
+warn_redundant_casts = true
+warn_unused_configs = true
+
+[build-system]
+requires = ["setuptools"]
+build-backend = "setuptools.build_meta"
+
+[tool.setuptools]
+packages = ["mache"]
```

Switching to the following worked for me:
```suggestion
[tool.setuptools.packages.find]
exclude = ["docs", "conda"]
```
(The exclusions are unnecessary because they don't contain a `__init__.py` but I just wanted to be explicit.)

You can see the discussion here:
https://setuptools.pypa.io/en/latest/userguide/package_discovery.html

I think we want to take advantage of automatic discovery.  Otherwise, we would have to list all the directories and subdirectories manually, e.g.:
```
[tool.setuptools]
packages = ["mache", "mache.machines", "mache.spack", "mache.sync"]
```
It would be nice if we could take advantage of automatic discovery and just say we want `mache` and all its contents but so far I don't see a way to do that.

---

I agree, let me try to explicitly search to see if there's some way to keep auto discovery on while also just using it from a specific root directory.

---

Let me know what you find out.

---

I'm not finding a way to use auto-discovery like we want to. It could be that I just can't find it, but since we have it working (with the changes you have here, I got them working as well), should we just leave it at that for now?

---

Okay, let's stick with what I came up with.  We can always "fix" it later.